registry/storage/catalog.go

```diff
@@ -5,12 +5,17 @@ import (
 	"errors"
 	"io"
 	"path"
+	"sort"
 	"strings"
 
 	"github.com/distribution/distribution/v3/reference"
 	"github.com/distribution/distribution/v3/registry/storage/driver"
 )
 
+var (
+	ErrStopRec = errors.New("Stopped the recursion for getting repositories")
```

Comment:
Can we add a comment about when is this error returned? I'd also like to change the error string to "stopped repository walk" and maybe call it something like `ErrWalkDone` or something like that?